app/controllers/katello/api/v2/module_streams_controller.rb

```diff
@@ -3,8 +3,35 @@ class Api::V2::ModuleStreamsController < Api::V2::ApiController
     apipie_concern_subst(:a_resource => N_("a module stream"), :resource => "module_streams")
     include Katello::Concerns::Api::V2::RepositoryContentController
 
+    before_action :check_hosts, :only => :index
+
+    update_api(:index) do
+      param :host_ids, Array, :desc => N_("List of host id to list available module streams for")
+    end
+
+    def index
+      super
+    end
```

Comment:
Do you still  need this index method? I would think its not necessary anymore with the addition of `update_api` method above